### PR TITLE
Dto+controller

### PR DESCRIPTION
**Pull request by forzen123212**

### CODE REVIEW

File: src/main/java/com/example/springfirsthw/repositories/BookRepositoryImpl.java [hunk 1]
```diff
@@ -45,12 +55,12 @@ public Book save(Book book) {
...
+    public Optional<Book> findById(Long id) {
         try (Session session = factory.openSession()) {
-            Query<Book> query = session.createQuery("from Book", Book.class);
-            return query.getResultList();
-        } catch (RuntimeException e) {
-            throw new RuntimeException(FAILED_TO_GET_ALL_BOOKS_FROM_DB_MSG);
+            return Optional.ofNullable(session.get(Book.class, id));
+        } catch (Exception e) {
+            throw new RuntimeException(
+                    FAILED_TO_FIND_BOOK_BY_ID);
```
Comment (BohdanTrue) [inline src/main/java/com/example/springfirsthw/repositories/BookRepositoryImpl.java]: You can specificate which id you failed to get. 
For example: `"Failed to find book by id: " + id, e`

File: src/main/java/com/example/springfirsthw/repositories/BookRepositoryImpl.java [hunk 2]
```diff
@@ -35,7 +45,7 @@ public Book save(Book book) {
             if (transaction != null) {
                 transaction.rollback();
             }
-            throw new RuntimeException(CANNOT_SAVE_BOOK_TO_DB_MSG + book, e);
+            throw new RuntimeException(CANNOT_SAVE_BOOK_TO_DB_MSG);
```
Comment (BohdanTrue) [inline src/main/java/com/example/springfirsthw/repositories/BookRepositoryImpl.java]: You can specificate which book you failed to save. 
For example: `"Failed to save book: " + book, e`

File: src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -1,27 +1,35 @@
...
-        return bookRepository.save(book);
+    public List<BookDto> getAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toBookDto)
+                .toList();
     }
 
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto getById(Long id) {
```
Comment (BohdanTrue) [inline src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java:33]: See, you have in Service layer `getById()` method, but in your Repository layer you have `findById()` method.
You know the difference between these two methods?

https://mate.academy/learn/java-spring-boot/java-spring-boot-web-dto?section=video&videoId=5560

File: src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -1,27 +1,35 @@
...
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("Can't find book by id"));
```
Comment (BohdanTrue) [inline src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java]: "Can't find book by id" -> "Can't find book by id: " + id, e 
and make constant variable

File: src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -1,27 +1,35 @@
...
 @Service
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    public BookDto save(CreateBookRequestDto book) {
+        Book bookSaved = bookRepository.save(bookMapper.toModel(book));
```
Comment (BohdanTrue) [inline src/main/java/com/example/springfirsthw/service/impl/BookServiceImpl.java]: make this inline
`return bookMapper.toBookDto(bookRepository.save(bookMapper.toModel(book)));`

File: src/main/java/com/example/springfirsthw/repositories/BookRepository.java [hunk 6]
```diff
@@ -2,13 +2,16 @@
...
+import java.util.Optional;
 
-@Repository
 public interface BookRepository {
 
     Book save(Book book);
 
     List<Book> findAll();
 
+    Optional<Book> findById(Long id);
```
Comment (BohdanTrue) [inline src/main/java/com/example/springfirsthw/repositories/BookRepository.java:13]: findById() or getById()? Let's think about it

**Review by BohdanTrue — CHANGES_REQUESTED** [review]
Good job! But have some points to consider!

File: src/main/java/com/example/springfirsthw/OnlineBookStoreApplication.java [hunk 7]
```diff
@@ -1,36 +1,17 @@
...
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
-import org.springframework.context.annotation.Bean;
 
+@RequiredArgsConstructor
 @SpringBootApplication
 public class OnlineBookStoreApplication {
 
-    @Autowired
     private BookService bookService;
```
Comment (vovasalyha) [inline src/main/java/com/example/springfirsthw/OnlineBookStoreApplication.java:12]: You don't need this sevice here anymore, so you can remove it.

File: src/main/java/com/example/springfirsthw/repositories/BookRepository.java (hunk 6 above)
Comment (vovasalyha) [reply src/main/java/com/example/springfirsthw/repositories/BookRepository.java:13]: By convention `findById` as it returns Optional

**Review by vovasalyha — APPROVED** [review]
Looks good, but remove unused service.

**Pull request merged**